source/Firmata/UwpFirmata.cpp

```diff
@@ -375,28 +318,28 @@ UwpFirmata::unlock(
 
 void
 UwpFirmata::inputThread(
-	void
-	)
+    void
+    )
 {
-	//set state-tracking member variables and begin processing input
-	while( !_input_thread_should_exit )
-	{
-		try
-		{
-			::RawFirmata.processInput();
-		}
-		catch( Platform::Exception ^e )
-		{
-			OutputDebugString( e->Message->Begin() );
-		}
-	}
+    //set state-tracking member variables and begin processing input
+    while( !_input_thread_should_exit )
+    {
+        try
+        {
+            ::RawFirmata.processInput();
+        }
+        catch( Platform::Exception ^e )
+        {
+            OutputDebugString( e->Message->Begin() );
+        }
+    }
 }
 
 
 void
 UwpFirmata::stopThreads(
-	void
-	)
+    void
+    )
 {
-	_input_thread_should_exit = true;
+    _input_thread_should_exit = true;
```

Comment:
The destructor calls finish() which joins this thread after calling stopThreads. The join logic could be moved to the stopThreads() function.